packages/react-dom/src/client/ReactDOMComponent.js

```diff
@@ -1343,3 +1352,157 @@ export function listenToEventResponderEventTypes(
+  bottom: number | void,
+  lastProps: Props,
+  hitSlopElement: HTMLElement,
+): void {
+  const hitSlopElementStyle = hitSlopElement.style;
+  if (lastProps.left !== left) {
+    if (left === undefined) {
+      hitSlopElementStyle.left = '';
+    } else {
+      hitSlopElementStyle.left = `-${left}px`;
+    }
+  }
+  if (lastProps.right !== right) {
+    if (right === undefined) {
+      hitSlopElementStyle.right = '';
+    } else {
+      hitSlopElementStyle.right = `-${right}px`;
+    }
+  }
+  if (lastProps.top !== top) {
+    if (top === undefined) {
+      hitSlopElementStyle.top = '';
+    } else {
+      hitSlopElementStyle.top = `-${top}px`;
+    }
+  }
+  if (lastProps.bottom !== bottom) {
+    if (bottom === undefined) {
+      hitSlopElementStyle.bottom = '';
+    } else {
+      hitSlopElementStyle.bottom = `-${bottom}px`;
+    }
+  }
+}
+
+export function handleEventTouchHitTarget(
+  lastProps: Props,
+  nextProps: Props,
+  rootContainerInstance: Container,
+  internalInstanceHandle: Object,
+  hostContext: HostContext,
+): void {
+  if (enableEventAPI) {
+    // Validates that there is a single element
+    const node = getElementFromTouchHitTarget(internalInstanceHandle);
+    if (node !== null) {
+      let parentNamespace: string;
+      if (__DEV__) {
+        const hostContextDev = ((hostContext: any): HostContextDev);
+        parentNamespace = hostContextDev.namespace;
+        warning(
+          parentNamespace === HTML_NAMESPACE,
+          'An event touch hit target was used in an unsupported DOM namespace. ' +
+            'Ensure the touch hit target is used in a HTML namespace.',
+        );
+      } else {
+        parentNamespace = ((hostContext: any): HostContextProd);
+      }
+
+      const element = ((node: any): HTMLElement);
+      // We update the event target state node to be that of the element.
+      // We can then diff this entry to determine if we need to add the
+      // hit slop element, or change the dimensions of the hit slop.
+      const lastElement = internalInstanceHandle.stateNode;
+      const left = nextProps.left;
+      const right = nextProps.right;
+      const top = nextProps.top;
+      const bottom = nextProps.left;
+
+      if (lastElement !== element) {
+        if (
+          left === undefined &&
+          right === undefined &&
+          top === undefined &&
+          bottom === undefined
+        ) {
+          return;
+        }
+        internalInstanceHandle.stateNode = element;
+        const hitSlopElement = createEventTargetHitSlop(
+          left,
+          right,
+          top,
+          bottom,
+          rootContainerInstance,
+          parentNamespace,
+        );
+        // We need to make the target relative so we can make the hit slop
+        // element inside it absolutely position around the target.
+        // TODO add a dev check for the computed style and warn if it isn't
+        // compatible.
+        element.style.position = 'relative';
```

Comment:
@necolas We need to come up with a good solution for this. We should mark this in the umbrella issue and track it as to how we can validate it and make sure stacking order etc is good.